vortex-layout/src/layouts/chunked/reader.rs

```diff
@@ -90,11 +90,15 @@ impl ChunkedReader {
                             )
                             .await?;
 
-                        Some(StatsTable::try_new(
-                            layout_dtype.clone(),
+                        if &stats_dtype != stats_array.dtype() {
+                            vortex_bail!("Expected stats DType {stats_dtype} doesn't match read stats dtype {}", stats_array.dtype())
+                        }
+
+                        // SAFETY: This is only fine to call because we perfrorm validation above
+                        Some(StatsTable::unchecked_new(
                             stats_array,
                             present_stats.into(),
-                        )?)
+                        ))
```

Comment:
We were recreating this stats dtype just so we can drop it in the constructor. Kept an old constructor and added a new one and explicitly checked for the dtype